hypothesis-python/src/hypothesis/searchstrategy/lazy.py

```diff
@@ -155,6 +155,9 @@ def __repr__(self):
     def do_draw(self, data):
         return data.draw(self.wrapped_strategy)
 
+    def do_filtered_draw(self, data, filter):
```

Comment:
`filter` is currently used in a few different ways:

- `filter.condition`
- `filter.not_satisfied` (which I'm fine with removing)
- `filter.note_retried` to emit a consistent message when rejection-sampling rejects for the first time
- `filter.default_do_filtered_draw` in `SearchStrategy.do_filtered_draw` (not strictly essential, but saves having to turn the default implementation into a static method or standalone function)

At the very least I think I'll rename it to `filter_strategy`, since `filtered_strategy` is unfortunately ambiguous.